File: UdidServer/src/horizon/util/DateUtil.java
package horizon.util;

import java.text.SimpleDateFormat;

public class DateUtil {
    private static final ThreadLocal<SimpleDateFormat> sDateFormatter = new ThreadLocal<SimpleDateFormat>() {
        @Override
        protected SimpleDateFormat initialValue() {
            return new SimpleDateFormat("yyyy-MM-dd HH:mm:ss:SSS");
        }
    };

    public static String now(){
       return sDateFormatter.get().format(System.currentTimeMillis());
    }
}

File: UdidServer/src/main/kotlin/horizon/util/DateUtil.kt
package horizon.util

import java.text.SimpleDateFormat

object DateUtil {
    private val sDateFormatter: ThreadLocal<SimpleDateFormat> = object : ThreadLocal<SimpleDateFormat>() {
        override fun initialValue(): SimpleDateFormat {
            return SimpleDateFormat("yyyy-MM-dd HH:mm:ss:SSS")
        }
    }

    fun now(): String {
        return sDateFormatter.get().format(System.currentTimeMillis())
    }
}
